aries_cloudagent/issuer/indy.py

```diff
@@ -234,29 +234,57 @@ def make_credential_definition_id(
 
         return credential_json, credential_revocation_id
 
-    async def revoke_credential(
-        self, revoc_reg_id: str, tails_file_path: str, cred_revoc_id: str
+    async def revoke_credentials(
+        self, revoc_reg_id: str, tails_file_path: str, cred_revoc_ids: Sequence[str]
     ) -> str:
         """
-        Revoke a credential.
+        Revoke a set of credentials in a revocation registry.
 
         Args:
             revoc_reg_id: ID of the revocation registry
             tails_file_path: path to the local tails file
-            cred_revoc_id: index of the credential in the revocation registry
+            cred_revoc_ids: sequences of credential indexes in the revocation registry
 
         Returns:
-            the revocation delta
+            the combined revocation delta
 
         """
         tails_reader_handle = await create_tails_reader(tails_file_path)
-        with IndyErrorHandler("Exception when revoking credential", IssuerError):
-            revoc_reg_delta_json = await indy.anoncreds.issuer_revoke_credential(
-                self.wallet.handle, tails_reader_handle, revoc_reg_id, cred_revoc_id
-            )
-            # may throw AnoncredsInvalidUserRevocId if using ISSUANCE_ON_DEMAND
 
-        return revoc_reg_delta_json
+        result_json = None
+        for cred_revoc_id in cred_revoc_ids:
+            with IndyErrorHandler("Exception when revoking credential", IssuerError):
+                # may throw AnoncredsInvalidUserRevocId if using ISSUANCE_ON_DEMAND
+                delta_json = await indy.anoncreds.issuer_revoke_credential(
+                    self.wallet.handle, tails_reader_handle, revoc_reg_id, cred_revoc_id
+                )
+                if not result_json:
```

Comment:
My two cents: if not ... else is a sign of overthinking.

Reply:
Maybe underthinking..